bin/ci-test

```diff
@@ -31,27 +33,72 @@ function handle_args() {
   fi
 }
 
+function build_core() {
+  set -x
+
+  docker build \
+    -t "dependabot/dependabot-core:latest" \
+    --build-arg BUILDKIT_INLINE_CACHE=1 \
+    --cache-from ghcr.io/dependabot/dependabot-core \
+    .
+}
+
+function build_core_ci() {
+  set -x
+
+  docker build \
+    -f Dockerfile.ci \
+    -t "dependabot-core-ci:latest" \
+    --build-arg BUILDKIT_INLINE_CACHE=1 \
+    .
+}
+
 function build() {
-  export DOCKER_BUILDKIT=1
-  export DOCKER_SCAN_SUGGEST=false
-  export CORE=dependabot/dependabot-core
-  export CI=dependabot/dependabot-core-ci
+  build_core
+  build_core_ci
+  run_suite
+}
 
+function run_suite() {
   set -x
-  docker build -qt $CORE .
-  docker build -qt $CI -f Dockerfile.ci .
-  docker run --rm \
-    -e "CI=true" \
-    -e "SUITE_NAME=$SUITE" \
-    -e "DEPENDABOT_TEST_ACCESS_TOKEN=${LOCAL_GITHUB_ACCESS_TOKEN}" \
-    -it $CI \
-    bash -c \
+
+  test -t 1 && USE_TTY="-it"
+
+  docker run $USE_TTY \
+    --env "CI=true" \
+    --env "RAISE_ON_WARNINGS=true" \
+    --env "DEPENDABOT_TEST_ACCESS_TOKEN=${LOCAL_GITHUB_ACCESS_TOKEN}" \
```

Comment:
I think we would want to do `env.` since the variable is set as a workflow environment variable in https://github.com/dependabot/dependabot-core/pull/5525/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fL69

This would probably break if you tried to run `run_suite` locally, though. Maybe we could test if the `LOCAL_GITHUB_ACCESS_TOKEN` is set first and use `env.` if it isn't?

```suggestion
    --env "DEPENDABOT_TEST_ACCESS_TOKEN=${env.LOCAL_GITHUB_ACCESS_TOKEN}" \
```


see https://docs.github.com/en/actions/learn-github-actions/environment-variables

Reply:
> Do the environment variables get passed to the workflow run commands?

> and the variables are interpolated on the runner machine that runs your workflow

Oh yep they totally are, disregard!